src/views/htmlcontent/src/js/components/app.component.ts

```diff
@@ -320,10 +323,25 @@ export class AppComponent implements OnInit, AfterViewChecked {
         });
         this.dataService.dataEventObs.subscribe(event => {
             switch (event.type) {
+                case 'start':
+                    self.uri = event.data;
+                    // Empty the data set if the query is run
+                    // again on the same panel
+                    if (self.hasRunQuery) {
+                        self.dataSets = [];
+                        self.placeHolderDataSets = [];
+                        self.renderedDataSets = self.placeHolderDataSets;
+                        self.messages = [];
```

Comment:
This is going to have these both pointing at the same object - is that intended? So changes made to one will also affect the other.